### DISCUSSION TITLE
네이버 잔여백신 새로고침 작동중에 에러가 발생합니다.

### DISCUSSION THREAD

**Discussion by yolocovid** [changdoc/naver-vaccine-macro#177] [Q&A]
안녕하세요. 당일 네이버 잔여백신 예약하고자 새로고침을 작동 시켰는데 네이버 지도 상 대기중이라는 문구만 발생하여 백신을 놓치는 경우도 있었고 (아예 페이지를 새로고침하니 0으로 바뀜) 네이버 지도 상 시간과 현재 시간이 안 맞는 부분이 있으며, 자꾸 에러가 뜹니다. 어떻게 해결해야 할까요? /(네이버에서 막아뒀는지 매크로를 돌리다가 안 돌아가서 끄고 다시 잔여백신 지도를 누르니 빈 페이지만 나오는 경우도 있었습니다. 확장 프로그램에서 개발자 모드 끄기로 하여 다시 접속하니 페이지가 생성됐습니다.)
![KakaoTalk_20210802_111215179](https://user-images.githubusercontent.com/88298935/127796423-066f14f8-b138-4807-8fce-d0b0fdb3da92.png)
![KakaoTalk_20210802_112236011](https://user-images.githubusercontent.com/88298935/127796424-8223c156-83bf-4adf-a167-3de393d7fd98.png)

**Comment by changdoc** [comment]
첫번째 스크린샷의 오류는 지도가 그려지기 전에 새로고침 버튼을 못찾아서 찍고 있는데 에러라 무시하셔도 괜찮을것 같습니다. (로그 레벨은 변경해두겠습니다.)

https://github.com/changdoc/naver-vaccine-macro/discussions/134
이 글 작성하신 분과 같은 상황인거 같은데 제가 겪어 보지 못한 에러라 원인 파악을 못했었는데요.
CORS이 발생하는것으로 봐서는 매크로를 삭제하고 업데이트를 눌러도 같은 에러가 발생하는지 한번 확인 해보는게 좋을거 같네요.

**Reply by yolocovid** [reply]
![headers 1](https://user-images.githubusercontent.com/88298935/127940672-bf3798e5-c88c-42c1-928e-7e74b7f2fa91.png)
![headers 2](https://user-images.githubusercontent.com/88298935/127940673-07179b56-3ab3-47aa-b347-844cedc6e12e.png)

Request URL: https://api.place.naver.com/graphql
Referrer Policy: unsafe-url
content-length: 587
content-type: text/html
date: Tue, 03 Aug 2021 00:37:38 GMT
server: nginx/1.14.2
:authority: api.place.naver.com
:method: POST
:path: /graphql
:scheme: https
accept: */*
accept-encoding: gzip, deflate, br
accept-language: ko
content-length: 2375
content-type: application/json
cookie: NNB=F6MWYMOF5F6V4; NRTK=ag#all_gr#1_ma#-2_si#0_en#0_sp#0; MM_NEW=1; NFS=2; ASID=daeb49a500000173333f1b3b0000006b; MM_NOW_COACH=1; nx_ssl=2; nid_inf=312659780; NID_AUT=7YATwf32ci1RKoNmGAndoHtF+C16VM9uC58OhCqaTBupew1tT+MeUBSVmv+OSmTL; NID_JKL=DXaruSjowIgsa/ySsMcGCdnek/Mx8KVehOANWVFCqUU=; page_uid=hdA5swprvhGssC32e6dssssstCN-424571; NID_SES=AAABk39+wyjSieZpCkPv4b2msdAoKurcaJXFhWtmZnzjRs9aReEouNU2H91uXZwL1hTPHSdNc0Hnd1cabyNHpImNIxRMCJfWO+44GOA5Y9dkXE9Sha5IVuhUJI9xrIFkCdNwnMbGf/nKR+fl2EySivXenDcqepfAJXFjZR8DJ/hqY+ny8NHdEAtooMgDGTNj2zK2rYJAEekZx0mBwo465OOCofhG+ypl7zeKYMqAdvmAu0NBgNm2M8t2EDMa3AQBBcFqdSLf2FfyUoBlWs7E1iUUWy/Kdti/aWSmDNAp6JefvLU13KMCFfMR2MyTKCmjGxqeqRwkcBZzn6w8euWIvs7DDJUe8tIWwov651Z3qHNR5x85zxKsbXiEppwkH8laEOSI1Ap1ZTexF5bYR39zG7YkTkZgAgWLRzjgAP/pP68DTVI/PFtxm2G0w2DEgK3/v90seKnwZv47G7ISIDqu8EBTdRSdT5EKSguiIAjwa/hwJUzYQ1cnm9qZteT8UPd7sPaeBId/tOwESWlHRHRebEWYz8H4QgRAfCVJnbJFlkO1IwN5
origin: https://m.place.naver.com
referer: https://m.place.naver.com/rest/vaccine?vaccineFilter=used&x=126.9528805&y=37.4805684&bounds=126.9416689%3B37.4749745%3B126.9640922%3B37.4861618
sec-ch-ua: "Chromium";v="92", " Not A;Brand";v="99", "Google Chrome";v="92"
sec-ch-ua-mobile: ?0
sec-fetch-dest: empty
sec-fetch-mode: cors
sec-fetch-site: same-site
user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36
[{operationName: "vaccineList",…}]
0: {operationName: "vaccineList",…}

**Reply by changdoc** [reply]
좌측에 빨간색 글씨 상태인 graphql도 동일할까요?

**Reply by changdoc** [reply]
올려주신건 정상인 요청에 대한 정보로 보이는데요

**Reply by yolocovid** [reply]
빨간 색은 위의 상단처럼 뜨고 

검은 색은 하단 처럼 뜹니다.
![Uploading headers black 1.png…]()



Request URL: https://api.place.naver.com/graphql
Request Method: OPTIONS
Status Code: 204 
Remote Address: 210.89.168.68:443
Referrer Policy: unsafe-url
access-control-allow-credentials: true
access-control-allow-headers: content-type
access-control-allow-methods: GET,HEAD,PUT,POST,DELETE,PATCH
access-control-allow-origin: https://m.place.naver.com
date: Tue, 03 Aug 2021 04:43:09 GMT
referrer-policy: unsafe-url
server: nfront
vary: Accept-Encoding, Origin
x-xss-protection: 1; mode=block
:authority: api.place.naver.com
:method: OPTIONS
:path: /graphql
:scheme: https
accept: */*
accept-encoding: gzip, deflate, br
accept-language: ko-KR,ko;q=0.9,en-US;q=0.8,en;q=0.7
access-control-request-headers: content-type
access-control-request-method: POST
origin: https://m.place.naver.com
referer: https://m.place.naver.com/rest/vaccine?vaccineFilter=used&x=126.9613056&y=37.4773129&bounds=126.9366507%3B37.4661242%3B126.9859604%3B37.4884999
sec-fetch-dest: empty
sec-fetch-mode: cors
sec-fetch-site: same-site
user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.131 Safari/537.36

**Reply by yolocovid** [reply]
![headers black 1](https://user-images.githubusercontent.com/88298935/127959275-eee9c017-cf40-4a3f-b3c9-e955bc4d7ac0.png)
![headers black 2](https://user-images.githubusercontent.com/88298935/127959280-9500b690-65f9-4dd4-9d05-36f161296536.png)

**Reply by yolocovid** [reply]
처음에는 잘 작동되다가 장시간 해두면 이렇게 바뀌어버리고 시간도 맞지 않습니다. 아무래도 네이버 쪽에서 무슨 장치를 해둔 것 같습니다. 이렇게 오류 뜬 뒤에 네이버 잔여백신 지도를 닫았다가 다시 열면 지도가 나타나지 않습니다.

**Reply by hyojik123** [reply]
저도 자꾸 빨간핑으로 뜨는데 잘 작동이안되네요

**Reply by changdoc** [reply]
음.. 저도 장시간 동작을 우선 시켜보도록 하겠습니다.
